content/en/docs/languages/ruby/basics.md

```diff
@@ -334,6 +334,34 @@ resps.each do |r|
 end
 ```
 
+Non-blocking usage of the RPC stream can be achieved with multiple threads and 
+the `return_op: true` flag. When passing the `return_op: true` flag, the 
+execution of the RPC is deferred and an `Operation` object is returned. The RPC 
+can then be executed in another thread by calling the operation `execute` 
+function. The main thread can utilize contextual methods and getters such as 
+`status`, `cancelled?`, and `cancel` to manage the RPC. This can be useful for 
+persistent or long running RPC sessions that would block the main thread for an
+unacceptable period of time.
+
+
+```ruby
+op = stub.list_features(LIST_FEATURES_RECT, return_op: true)
+Thread.new do 
+  resps = op.execute
+  resps.each do |r|
+    p "- found '#{r.name}' at #{r.location.inspect}"
+  end
+rescue GRPC::Cancelled => e
+  p "operation cancel called - #{e}"
+end
+
+# controls for the operation
+op.status
+op.cancelled?
+op.cancel # terminates the connection and raises GRPC::Cancelled in the blocking thread if successful.
```

Comment:
nit: the "if successful" wording here is still a bit confusing.

I think we need to make clear the following:
- this will attempt to cancel the RPC with a GRPC::Cancelled status
- there's a fundamental race condition where cancelling the RPC can race against RPC termination for a different reason. So invoking `cancel` doesn't necessarily guarantee a `Cancelled` status